File: WeekendContest/LeetCode_2466.cpp
// 2466. Count Ways To Build Good Strings

class Solution {
public:
    int countGoodStrings(int low, int high, int zero, int one) {
        vector<long> v(high+1, 0L); v[zero] += 1; v[one] += 1;
        long ans=0, mod = 1e9+7;
        for (int i=0; i<high+1; ++i){
            v[i] = (v[i] + ((i-zero > 0)? v[i-zero]: 0) + ((i-one > 0)? v[i-one]: 0)) % mod;
            if (i>= low)
                ans = (ans + v[i]) % mod;
        }
        return ans;
    }
};

File: DailyChallenges/Scala/LeetCode_2466.scala
object Solution {
    def countGoodStrings(low: Int, high: Int, zero: Int, one: Int): Int = {
        val dp = Array.fill[Int](high+1)(0); dp(zero) += 1; dp(one) += 1
        val mod = 1000000007
        var ans = 0
        for (i <- 0 to high){
            if (i-zero>0) dp(i) = (dp(i) + dp(i-zero)) % mod
            if (i-one>0) dp(i) = (dp(i) + dp(i-one)) % mod
            if (i>=low) ans = (ans + dp(i)) % mod
        }
        ans
    }
}
